e2eAIOK/common/trainer/utils/utils.py

```diff
@@ -200,9 +274,126 @@ def create_metric(cfg):
 def create_criterion(cfg):
     if cfg.criterion == "CrossEntropyLoss":
         criterion = torch.nn.CrossEntropyLoss()
+    else:
+        logging.error("[%s] is not supported"%cfg.criterion)
+        raise NotImplementedError("[%s] is not supported"%cfg.criterion)
     return criterion
 
+def create_optimizer(model=None, cfg=None, parameters=None):
+    ''' create optimizer
+
+    :param model: model, will be disabled if 'parameters' is not None
+    :param cfg: configurations
+    :param parameters: parameters for optimizer
+
+    :return: a optimizer
+    '''
+    print(F"model:{model}")
+    print(F"parameters:{parameters}")
```

Comment:
please use logging to for easier log tracking

Reply:
have modified.